Allow overlapping plate boundaries and networks

cpalfonso opened the pull request:
Subduction zone segments are now allowed to border up to four topologies - one plate boundary and one deforming network on each side. Velocities should be preferentially taken from networks.

This means that the subduction convergence workflow is now compatible with the newer policy of having the rigid plate boundary topologies cover the entire globe and then placing the deforming networks on top of those topologies.

jcannon-gplates commented:
Thanks for this. Just a note that the subduction convergence is moving towards sampling the trench plate ID instead of the overriding plate ID (you can see that with pygplates version >= 0.22, where the overriding plate ID isn't used at all).

So since the public release of pyGPlates 0.28 has been out since 2020 (and actually a new public release of PyGPlates 0.36 is coming out any day now), and you've made changes to remove unused pyGPlates versions, it might be good to remove the code functions/paths that use the overriding plate ID (based around `USING_PYGPLATES_VERSION_GREATER_EQUAL_22`) and increase our minimum requirement to 0.28 and have that be the only version hardcoded (ie, `PYGPLATES_VERSION_REQUIRED`).

And then I think your contribution will be to allow the subducting side to have a single rigid plate and a single deforming network (with preference given to the network)?  Instead of enforcing a single boundary *or* network.  Actually I guess there's still the issue of whether a network has a valid plate ID (they're not required to, unlike a rigid plates). I guess I had assumed that *subducting* networks would be quite rare (I vaguely recall chatting with Sabin about this) - currently those regions would get discarded so your change at least avoids that (which is good). Ultimately we should get the network boundary velocity from pyGPlates (once it supports that query). For now I guess networks tend to have a plate ID assigned in our topological models (you might know that better than me) ?

Also if you could perhaps update the code comments to reflect your changes, eg, just a few comments that are now a little outdated.

cpalfonso commented:
I've changed the `find_subducting_plate` function so that it now only takes the plate ID from a deforming network if the network has been assigned a plate ID – otherwise, it will fall back to the rigid plate boundary topology. I've also changed the minimum pyGPlates version to 0.28 and removed the references to earlier versions.

jcannon-gplates commented:
Thanks, looks good!
